builtin/credential/jwt/path_config_test.go

```diff
@@ -745,6 +772,208 @@ func TestConfig_OIDC_Ignore(t *testing.T) {
 	}
 }
 
+// certificate is valid for *.badssl.com, badssl.com, not wrong.host.badssl.com
+func TestConfig_CAContext_MismatchedHost(t *testing.T) {
+	type testCase struct {
+		nameInCertificate string
+		allowedServerNames []string
+		expectedFailureString string
+		expectSuccess bool
+		addRootCA bool
+	}
+	tests := map[string]testCase{
+		"domain not in list": {
+			nameInCertificate: "example.com",
+			allowedServerNames: []string{"invalid.com"},
+			addRootCA: true,
+			expectSuccess: false,
+			expectedFailureString: "certificate is valid for example.com, not invalid.com",
+		},
+		"invalid root CA": {
```

Comment:
FYI @cipherboy added a specific check to ensure overriding RootCAs in combination with allowedServerNames is working correctly. I've also removed any dependencies to badssl.